projects/libs/core-library-typescript/src/common-library/transaction-utils.ts

```diff
@@ -0,0 +1,15 @@
+import { types } from "../common-library/types";
+import { createHash } from "crypto";
+
+export namespace TransactionUtils {
+  export function calculateTransactionHash(transaction: types.Transaction): Buffer {
+    const hash = createHash("sha256");
+    hash.update(JSON.stringify(transaction.header));
```

Comment:
Can we order these objects somehow? Because this hash is non-deterministic and we can't rely on it, all network comes to a halt after a very short time.

Reply:
Maybe use something like this https://github.com/substack/json-stable-stringify or write our own implementation.

Reply:
This hashing mechanism is *very temporary*. Basically we're going to change it as soon as we implement the transaction signature field

Reply:
I understand it's temporary, but the problem is that some transactions will inevitably fail to clear out of transaction polls on some nodes.

Reply:
You're right. This is another issue. I'll make it deterministic